interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -59,32 +59,36 @@ def create_casadi_nlp_formulation(cls, ocp: AcadosOcp) -> Tuple[dict, dict]:
         if any([dims.ns_0, dims.ns, dims.ns_e]):
             raise NotImplementedError("CasADi NLP formulation not implemented for formulations with soft constraints yet.")
 
-        # create variables
+        # create variables indexed by shooting nodes, vectorize in the end
         ca_symbol = model.get_casadi_symbol()
         xtraj = ca_symbol('x', dims.nx, solver_options.N_horizon+1)
+        xtraj_node = [xtraj[:, i] for i in range(xtraj.shape[-1])]
```

Comment:
maybe use `ca.horzsplit`

Reply:
I actually think not creating a matrix at all is preferred, like just 1 line instead of 2:
`xtraj_node = [ca_symbol(f'x{i}', dims.nx, 1) for i in range(N_horizon+1)]`

Because
a) extensibility for varying dimension (multi-phase and slacks)
b) As the x values are only used individually, I would bet the resulting solver to be more efficient if the x values are individual symbols. (Not verified though).